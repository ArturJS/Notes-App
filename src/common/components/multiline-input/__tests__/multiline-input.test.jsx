import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MultilineInput from '../multiline-input';

describe('<MultilineInput />', () => {
    beforeEach(() => {
        jest
            .spyOn(window, 'requestAnimationFrame')
            .mockImplementation(cb => cb());
    });

    afterEach(() => {
        window.requestAnimationFrame.mockRestore();
    });

    it('renders correctly', () => {
        const props = {
            input: {
                onChange: () => {},
                value: 'test value'
            },
            className: 'test-class-name',
            placeholder: 'test placeholder'
        };
        const tree = renderer.create(<MultilineInput {...props} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should invoke `autofitContent` method after `componentDidMount`', () => {
        const props = {
            input: {
                onChange: () => {},
                value: 'test value'
            },
            className: 'test-class-name',
            placeholder: 'test placeholder'
        };
        const wrapper = mount(<MultilineInput {...props} />);
        const instance = wrapper.instance();

        jest.spyOn(instance, 'autofitContent');
        instance.componentDidMount();

        expect(instance.autofitContent).toHaveBeenCalled();
        expect(window.requestAnimationFrame).toHaveBeenCalled();
    });

    it('should invoke `autofitContent` method after `componentDidUpdate` if no value', () => {
        const props = {
            input: {
                onChange: () => {},
                value: 'test value'
            },
            className: 'test-class-name',
            placeholder: 'test placeholder'
        };
        const wrapper = mount(<MultilineInput {...props} />);

        wrapper.setProps({
            ...props,
            input: {
                onChange: () => {},
                value: ''
            }
        });

        const instance = wrapper.instance();

        jest.spyOn(instance, 'autofitContent');
        instance.componentDidUpdate();

        expect(instance.autofitContent).toHaveBeenCalled();
        expect(window.requestAnimationFrame).toHaveBeenCalled();
    });
});
