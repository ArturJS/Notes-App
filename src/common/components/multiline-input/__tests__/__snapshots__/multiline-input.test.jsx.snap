// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultilineInput /> renders correctly 1`] = `
<textarea
  className="multiline-input test-class-name"
  onChange={[Function]}
  onInput={[Function]}
  placeholder="test placeholder"
  value="test value"
/>
`;
